app/api/check_cumulative.py

```diff
@@ -0,0 +1,109 @@
+"""
+Check for cumulative data decreases from week to week.
+"""
+import csv
+from time import time
+
+import flask
+import pandas as pd
+
+from app.api import utils
+
+def do_check_cumulative(outputFile):
+    states = utils.get_all_state_finals()
+
+    with open(outputFile, mode='w') as csv_file:
+        fieldnames = ['State', 'Facility', 'Date', 'cume_res_pos_value', 'prev_cume_res_pos_value']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+        writer.writeheader()
+
+        def check_cumulative(df):
+            collection_dates = df[['Date']].drop_duplicates()
+
+            last_cumes = {}
+            last_cume = 0
+
+            for date_index, date_row in collection_dates.iterrows():
```

Comment:
since you're not using `date_index`, a bit simpler to do:
```
collection_dates = set(df['Date'])
for collection_date in collection_dates:
   current_block = df.loc[df['Date'] == collection_date]
   ...
```

Reply:
👍